benchmarks/differenciation/melt_srb.cc

```diff
@@ -0,0 +1,698 @@
+                                                                          - in.composition[i][peridotite_idx] * trace(in.strain_rate[i]);
+                              else if (c == porosity_idx && this->get_timestep_number() > 0)
+                                reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale;
+                              else
+                                reaction_rate_out->reaction_rates[i][c] = 0.0;
+                            }
+                          out.reaction_terms[i][c] = 0.0;
+                        }
+                    }
+
+                  // find depletion = peridotite, which might affect shear viscosity:
+                  const double depletion_visc = std::min(1.0, std::max(in.composition[i][peridotite_idx],0.0));
+
+                  // calculate strengthening due to depletion:
+                  const double depletion_strengthening = std::min(exp(alpha_depletion*depletion_visc),delta_eta_depletion_max);
+
+                  // calculate viscosity change due to local melt and depletion:
+                  out.viscosities[i] *= depletion_strengthening;
+                }
+            }
+
+          out.entropy_derivative_pressure[i]    = 0.0;
+          out.entropy_derivative_temperature[i] = 0.0;
+          out.thermal_expansion_coefficients[i] = thermal_expansivity;
+          out.specific_heat[i] = reference_specific_heat;
+          out.thermal_conductivities[i] = thermal_conductivity;
+          out.compressibilities[i] = 0.0;
+
+          double visc_temperature_dependence = 1.0;
+          if (this->include_adiabatic_heating ())
+            {
+              const double delta_temp = in.temperature[i]-this->get_adiabatic_conditions().temperature(in.position[i]);
+              visc_temperature_dependence = std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/this->get_adiabatic_conditions().temperature(in.position[i])),1e4),1e-4);
+            }
+          else if (thermal_viscosity_exponent != 0.0)
+            {
+              const double delta_temp = in.temperature[i]-reference_T;
+              visc_temperature_dependence = std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/reference_T),1e4),1e-4);
+            }
+          out.viscosities[i] *= visc_temperature_dependence;
+        }
+
+      // fill melt outputs if they exist
+      MeltOutputs<dim> *melt_out = out.template get_additional_output<MeltOutputs<dim> >();
+
+      if (melt_out != nullptr)
+        {
+          const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+          for (unsigned int i=0; i<in.position.size(); ++i)
+            {
+              double porosity = std::max(in.composition[i][porosity_idx],0.0);
+
+              melt_out->fluid_viscosities[i] = eta_f;
+              melt_out->permeabilities[i] = reference_permeability * std::pow(porosity,2); // * std::pow(1.0-porosity,2);
```

Comment:
Please remove the commented code, unless it is important. Leaving commented code around is bad style and causes confusion in the future (plus we do not need it, because we have git for the history).

Reply:
done (and I added info because this is one of the places where I modified the initial melt model)